browser/ui/split_view/split_view_link_navigation_throttle.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/split_view/split_view_link_navigation_throttle.h"
+
+#include "base/feature_list.h"
+#include "brave/browser/ui/split_view/split_view_link_redirect_utils.h"
+#include "chrome/browser/ui/ui_features.h"
+#include "content/public/browser/navigation_handle.h"
+#include "ui/base/window_open_disposition.h"
+
+// static
+void SplitViewLinkNavigationThrottle::MaybeCreateAndAdd(
+    content::NavigationThrottleRegistry& registry) {
+  if (!base::FeatureList::IsEnabled(features::kSideBySide)) {
+    return;
+  }
+
+  registry.AddThrottle(
+      std::make_unique<SplitViewLinkNavigationThrottle>(registry));
+}
+
+SplitViewLinkNavigationThrottle::SplitViewLinkNavigationThrottle(
+    content::NavigationThrottleRegistry& registry)
+    : content::NavigationThrottle(registry) {}
+
+SplitViewLinkNavigationThrottle::~SplitViewLinkNavigationThrottle() = default;
+
+content::NavigationThrottle::ThrottleCheckResult
+SplitViewLinkNavigationThrottle::WillStartRequest() {
+  return MaybeRedirectToRightPane();
+}
+
+content::NavigationThrottle::ThrottleCheckResult
+SplitViewLinkNavigationThrottle::MaybeRedirectToRightPane() {
+  content::WebContents* source = navigation_handle()->GetWebContents();
+  if (!source) {
+    return PROCEED;
+  }
+
+  // Only handle main frame navigations
+  if (!navigation_handle()->IsInMainFrame()) {
+    return PROCEED;
+  }
+
+  // Only intercept user-initiated navigations.
+  if (!navigation_handle()->HasUserGesture()) {
+    return PROCEED;
+  }
+
+  // Only intercept renderer-initiated navigations.
+  if (!navigation_handle()->IsRendererInitiated()) {
+    return PROCEED;
+  }
+
+  // Don't intercept POST requests (forms)
+  if (navigation_handle()->IsPost()) {
```

Comment:
Out of curiosity, should we care about other http request methods like `PUT`, `PATCH` here too? If we shouldn't intercept them, we would want this to be `GetRequestMethod() != 'GET'`